GuideFinder-Mac/GuideFinder_Mac_2020_SMM_PAMless.Rmd

```diff
@@ -0,0 +1,1365 @@
+}
+
+NewCoordinatesDF<-NewCoordinatesDF2 #renaming 
+
+
+```
+
+
+__Pulls CDS + promoter from coordinates__  
+Makes the coordinates numeric so they can be used to identify location in FASTA file and pull desired CDS and putative promoter. Adds a unique ID # to each gene. For genes coded on the minus strand, takes the reverse complement to put the sequence in the proper orientation required by the rest of the program. Subsets the promoter from the pulled sequence (the additional # of bp, defined in PromoterRegion). 
+```{r, eval=FALSE,tidy=TRUE}
+#making coordinates numeric 
+NewCoordinatesDF$start<-as.numeric(as.character(NewCoordinatesDF$start)) 
+NewCoordinatesDF$end<-as.numeric(as.character(NewCoordinatesDF$end) )
+
+#loop to pull all CDS given new coordinates 
+CodingSequences<-NULL 
+for (i in 1:nrow(NewCoordinatesDF)) { 
+  myRow<-NewCoordinatesDF[i,] 
+  StartCDS<-myRow$start #set start coordinate as StartCDS 
+  EndCDS<-myRow$end #set end coordinate as EndCDS 
+  #Pulls CDS from input genome (defined above), using StartCDS and EndCDS 
+  Sequence<-InputGenome[[1]][StartCDS:EndCDS]
+  newDF<-data.frame(StartCDS, EndCDS,as.character(Sequence),
+                    myRow$geneproduct, myRow$strand ) #convert the sequence to character 
+  CodingSequences<-rbind(CodingSequences, newDF) 
+}
+
+
+CodingSequences$ID<-1:nrow(CodingSequences) #Add Unique ID number to each gene  
+names(CodingSequences)<-c("StartCDS", "EndCDS", "CDS", "Product", "Strand", "ID")  
+CodingSequences<-replace(CodingSequences,3, as.character(CodingSequences$CDS))
+
+#Take the reverse complement of the coding sequences designated on the "minus" strand  
+CorrectedCodingSequences<-NULL
+for (i in 1:nrow(CodingSequences)) {
+  NewRow<-CodingSequences[i,]
+  #runs PlusorMinus function (defined in function section) to ask strand type 
+  StrandType<-PlusorMinus(NewRow$Strand)
+  #if the strand type is minus, will rev-comp the coding sequence to put in proper orientation  
+  testing<-if (StrandType == "MINUS") 
+    { replace(NewRow,3, str_c(rev(comp(s2c(NewRow$CDS))), collapse = ""))} 
+  else {replace(NewRow,3,NewRow$CDS)} #if strand type is positive, do nothing
+  CorrectedCodingSequences<-rbind(CorrectedCodingSequences, testing)
+}
+
+
+CorrectedCodingSequences<-replace(CorrectedCodingSequences, 3 ,
+                                  unlist(lapply(CorrectedCodingSequences$CDS,
+                                                toupper))) #make all sequences uppercase 
+
+#create a dataframe that contains only the coding sequence and the gene product 
+OnlySequences<-data.frame(CorrectedCodingSequences$CDS, 
+                          CorrectedCodingSequences$Product,  
+                          CorrectedCodingSequences$ID)
+names(OnlySequences)<-c("CDS", "GeneProduct","ID")
+OnlySequences$CDS<-as.character(OnlySequences$CDS) 
+OnlySequences$Promoter<-lapply(OnlySequences$CDS,SubsetPromoter) #subset promoter 
+```
+
+
+__Finding PAM sites__  
```

Comment:
Maybe clarify that for PAMless, this chunk should still be run and that the PAMless_guides parameter set above will modify the FindCC function to find all sites, but that his chunk creates a necessary dataframe used in the next chunk.. as long as I am not mistaken about this.. otherwise a note not to run this if PAMless_guieds == true? or maybe just a chunk name change to Guide sites? 
```suggestion
__Finding PAM and/or PAMless sites__  
```